content/blog/2022/2022-08-25-schedule-cron-job-in-node.md

```diff
@@ -0,0 +1,277 @@
+---
+title: "Schedule Job On Node.js"
+categories: ["Node"]
+date: 2022-08-25 00:00:00 +1100
+authors: ["ajibade"]
+description: "Quick guide to scheduling cron jobs in Node.js applications"
+image: images/stock/0031-matrix-1200x628-branded.jpg
+url: schedule-cron-job-in-node
+---
+
+Have you ever wanted to perform a specific task on your application server at specific times without physically running them yourself? Or you'd rather spend your time on more important tasks than remembering to periodically clear or move data from one part of the server to another. Cron Job schedulers can be used to achieve this easily.
+
+Cron job scheduling is a common practice in applications; it provides a level of autonomy in our applications.
+
+## Using node cron In node.Js
+In this article, we'll use the [node-cron](https://www.npmjs.com/package/node-cron) library to schedule cron jobs in a Node.js application.
+
+Cron is a [daemon](https://en.wikipedia.org/wiki/Daemon_(computing)), which means that it only needs to be started once, and will 
+lay dormant until it is required. Another example of deamon is the web server, it remains idle until a request for a web page is made. The cron daemon remains idle until the time specified in its configuration files, or crontabs is reached.
+
+
+### Node cron syntax
+```
+cron.schedule(expression , function, options);
+```
+- **The first argument** passed to the node-cron syntax is a cron-expression (crontab). This expression is used to specify when a cron job is to be executed.
+
+    The cron expression used here can be represented using Asterix  (*),  which are wildcard characters representing the scheduled time for cron jobs. The expression by default will be in the format `* * * * * *`.
+
+Here's a quick reference to the cron expression, with asterisks indicating what each one represents.
+
+```
+ # ┌────────────── second (0 - 59) (optional)
+ # │ ┌──────────── minute (0 - 59) 
+ # │ │ ┌────────── hour (0 - 23)
+ # │ │ │ ┌──────── day of the month (1 - 31)
+ # │ │ │ │ ┌────── month (1 - 12)
+ # │ │ │ │ │ ┌──── day of the week (0 - 6) (0 and 7 both represent Sunday)
+ # │ │ │ │ │ │
+ # │ │ │ │ │ │
+ # *  * * * * *  command to execute
```

Comment:
I guess this snippet comes from some `crontab` docs? Since `node-cron` is not a command-line tool there is no `command to execute`, is there?